.github/workflows/runtests.yml

```diff
@@ -114,35 +114,23 @@ jobs:
       - run: bin/doctest --force-colors
       - run: examples/all.py -q
 
-  # -------------------- Test split 1/2 latest Python -------------- #
+  # ------------------------ Test latest Python -------------------- #
 
-  tests1-latest:
-    needs: code-quality
-
-    runs-on: ubuntu-20.04
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
-        with:
-          python-version: '3.10'
-      - run: python -m pip install --upgrade pip
-      - run: pip install mpmath
-      - run: bin/test --force-colors --split=1/2
-
-  # -------------------- Test split 2/2 latest Python -------------- #
-
-  tests2-latest:
+  tests-latest:
     needs: code-quality
 
     runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        group: [1, 2, 3, 4]
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
         with:
           python-version: '3.10'
       - run: python -m pip install --upgrade pip
-      - run: pip install mpmath
-      - run: bin/test --force-colors --split=2/2
+      - run: pip install mpmath pytest pytest-split pytest-xdist
+      - run: pytest -m 'not slow' -n auto --splits 4 --group ${{ matrix.group }}
```

Comment:
Maybe `bin/test` could be changed to be a wrapper script that translates common arguments to the pytest equivalent e.g. it could give `-m 'not slow'` automatically.